### PR TITLE
ephemeral volume tests

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

SIG-Storage wants to move ephemeral inline CSI volumes to beta in 1.16. These tests cover various aspects of that feature.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by pohly** [comment]
@kfox1111, @vladimirvivien, @msau42: the API change was merged (https://github.com/kubernetes/kubernetes/pull/80568). Here are additional E2E tests for the feature.

Initially I also had the API unit test enhancement in this PR, but then moved it into https://github.com/kubernetes/kubernetes/pull/81961 because that part is less important and would need different approvers.

**Comment by pohly** [comment]
The alpha feature testing worked fine in https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81960/pull-kubernetes-e2e-gce-alpha-features/1166211517989785601/#0:build-log.txt%3A2422:
* [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: inline ephemeral CSI volume] ephemeral [Feature:CSIInlineVolume] should support two pods which share the same volume
* [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: inline ephemeral CSI volume] ephemeral [Feature:CSIInlineVolume] should create read/write inline ephemeral volume
* [sig-storage] CSI mock volume CSI workload information using mock driver contain ephemeral=true when using inline volume [Feature:CSIInlineVolume]
* [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: inline ephemeral CSI volume] ephemeral [Feature:CSIInlineVolume] should create read-only inline ephemeral volume
* [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: inline ephemeral CSI volume] ephemeral [Feature:CSIInlineVolume] should support multiple inline ephemeral volumes

**Comment by pohly** [comment]
pull-kubernetes-conformance-kind-ipv6 failing is a known issue (https://kubernetes.slack.com/archives/C9NK9KFFW/p1566841675142300), fix pending in https://github.com/kubernetes/kubernetes/pull/81750.

**Review by vladimirvivien — COMMENTED** [review]
Looks good. Left couple minor comments.

File: test/e2e/storage/csi_mock_volume.go [hunk 1]
```diff
@@ -759,19 +766,23 @@ func checkPodInfo(cs clientset.Interface, namespace, driverPodName, driverContai
...
-		if call.Method != "/csi.v1.Node/NodePublishVolume" {
-			continue
-		}
-		// Check that NodePublish had expected attributes
-		for k, v := range expectedAttributes {
-			vv, found := call.Request.VolumeContext[k]
-			if found && v == vv {
-				foundAttributes.Insert(k)
-				e2elog.Logf("Found volume attribute %s: %s", k, v)
+		switch call.Method {
```
[the remark this answers is not in the document]
Comment (pohly) [reply test/e2e/storage/csi_mock_volume.go]: Correct. The approach itself isn't new, though, and seems to have worked so far.

File: test/e2e/testing-manifests/storage-csi/external-resizer/rbac.yaml [hunk 2]
```diff
@@ -36,9 +36,6 @@ rules:
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
     verbs: ["update", "patch"]
-  - apiGroups: ["storage.k8s.io"]
```
[the remark this answers is not in the document]
Comment (pohly) [reply test/e2e/testing-manifests/storage-csi/external-resizer/rbac.yaml]: Yes. The copy of the file in Kubernetes was out-of-sync with the upstream copy. Now the copy is identical with the original (= https://github.com/kubernetes-csi/external-resizer/blob/v0.2.0/deploy/kubernetes/rbac.yaml).

**Comment by pohly** [comment]
/retest

Not sure whether pull-kubernetes-conformance-kind-ipv6 fails randomly - let's try again...

**Comment by pohly** [comment]
/retest

https://github.com/kubernetes/kubernetes/pull/81750 still stuck in the merge pool for two hours. But other PRs have passed pull-kubernetes-conformance-kind-ipv6,  so perhaps we just need to keep retrying...

**Comment by pohly** [comment]
Force-pushed a rebased version. Some utility code was moved into `test/e2e/framework/pod`.

**Comment by pohly** [comment]
I resolved another conflict (external-resizer landed in master).

**Pull request merged**